servo/connectors/kubernetes.py

```diff
@@ -2539,9 +3124,14 @@ class CanaryOptimization(BaseOptimization):
 
     @classmethod
     async def create(
-        cls, deployment_config: "DeploymentConfiguration", **kwargs
+        cls, deployment_config: Union["DeploymentConfiguration", "RolloutConfiguration"], **kwargs
     ) -> "CanaryOptimization":
-        deployment = await Deployment.read(deployment_config.name, cast(str, deployment_config.namespace))
+        if isinstance(deployment_config, DeploymentConfiguration):
+            deployment = await Deployment.read(deployment_config.name, cast(str, deployment_config.namespace))
+        elif isinstance(deployment_config, RolloutConfiguration):
+            deployment = await Rollout.read(deployment_config.name, cast(str, deployment_config.namespace))
```

Comment:
Don't overload the name `deployment` like this. Abstract it to something like `target` or even `deployment_or_rollout` so when you read it later you don't falsely assume that it is a Deployment